File: fib.go
// 698 ms

package main

import (
	"fmt"
	"time"
)

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	t0 := time.Now()
	fib(40)
	fmt.Println(time.Now().Sub(t0))
}

File: fib.clj
(defn fib
  ([n]
    (fib [1 1] n))
  ([x, n]
    (if (< (count x) n)
      (fib (conj x (+ (last x) (nth x (- (count x) 2)))) n)
      x)))

; alternative way using lazy-seq
; (defn fib [a b] (cons a (lazy-seq (fib b (+ b a)))))

(time (fib 40))
